src/scenes/Authentication/Register/Register.tsx

```diff
@@ -0,0 +1,53 @@
+import { FORM_ERROR } from 'final-form';
+import React, { useEffect, useState } from 'react';
+import { useNavigate, useSearchParams } from 'react-router-dom';
+import { Except } from 'type-fest';
+import { handleFormError, useRegisterMutation } from '../../../api';
+import { useSession } from '../../../components/Session';
+import { RegisterFormProps as Props, RegisterForm } from './RegisterForm';
+
+export const Register = (props: Except<Props, 'onSubmit'>) => {
+  const [register] = useRegisterMutation();
+  const navigate = useNavigate();
+  const query = useSearchParams();
+  const [session, sessionLoading, setUserSession] = useSession();
+  const [success, setSuccess] = useState(false);
+
+  // Redirect to homepage if already logged in (and not from successful login)
+  useEffect(() => {
+    if (!sessionLoading && session && !success) {
+      navigate('/', { replace: true });
+    }
+  }, [navigate, session, sessionLoading, success]);
+
+  const submit: Props['onSubmit'] = async (input) => {
+    try {
+      const { data } = await register({
+        variables: {
+          input: {
+            user: {
+              displayFirstName: input.firstName,
+              displayLastName: input.lastName,
+              realFirstName: input.firstName,
+              realLastName: input.lastName,
+              email: input.email,
+              password: input.password,
+            },
+          },
+        },
+      });
+      setSuccess(true);
+      setUserSession(data.createUser.user);
+      const returnTo = decodeURIComponent(query.get('returnTo') ?? '/');
+      navigate(returnTo, { replace: true });
+    } catch (e) {
+      return await handleFormError(e, {
+        Default: {
+          [FORM_ERROR]: `Something wasn't right. Try again, or reset password.`,
+        },
+      });
```

Comment:
Error handlers can just return a string to indicate the same thing.
```diff
- Default: {
-   [FORM_ERROR]: `Something wasn't right. Try again.`,
- },
+ Default: `Something wasn't right. Try again.`,
```
However, I'm also thinking we should probably not copy/paste this generic error string everywhere. Since we aren't handling a specific case here let's just leave it off. Currently that means that the server message will be used. Later if we want to replace that with a generic message we can do that in the `handleFormErrors` function.
```suggestion
      return await handleFormError(e);
```

Reply:
Perfect 👍